imap-proto/src/parser/rfc3501.rs

```diff
@@ -97,6 +97,12 @@ named!(resp_text_code_alert<ResponseCode>, do_parse!(
     (ResponseCode::Alert)
 ));
 
+named!(resp_text_code_capability<ResponseCode>, do_parse!(
+    tag_s!("CAPABILITY") >>
+    capabilities: many1!(capability) >>
```

Comment:
It looks like technically this requires that the list of capabilities includes `IMAP4rev1`, maybe we should error out if it does not?

Reply:
I'm not sure such a test belongs to the parser. The client may want to try using a server that doesn't include IMAP4rev1 in its capabilities.

Reply:
The formal syntax in the RFC requires that it's there, so I think that makes it a parser-level check, wouldn't you agree?

Reply:
Indeed, I hadn't noticed it was part of the grammar.